packages/office-ui-fabric-react/src/components/DetailsList/DetailsColumn.tsx

```diff
@@ -0,0 +1,255 @@
+      setTimeout(() => {
+        if (this._root!.current!) {
+          this._root!.current!.classList!.remove(styles.borderAfterDropping);
+        }
+      }, 1500);
+    }
+  }
+
+  public componentWillUnmount(): void {
+    if (this._dragDropSubscription) {
+      this._dragDropSubscription.dispose();
+      delete this._dragDropSubscription;
+    }
+  }
+
+  public componentDidUpdate(): void {
+    if (!this._dragDropSubscription && this.props.dragDropHelper && this.props.isDraggable!) {
+      this._dragDropSubscription = this.props.dragDropHelper.subscribe(
+        this._root.value as HTMLElement,
+        this._events,
+        this._getColumnDragDropOptions()
+      );
+
+      // We need to use native on this to avoid MarqueeSelection from handling the event before us.
+      this._events.on(this._root.current, 'mousedown', this._onRootMouseDown);
+    }
+    if (this._dragDropSubscription && !this.props.isDraggable!) {
+      this._dragDropSubscription.dispose();
+      // We need to use native on this to avoid MarqueeSelection from handling the event before us.
+      this._events.off(this._root.current, 'mousedown');
+      delete this._dragDropSubscription;
+    }
+  }
+
+  private _onRenderColumnHeaderTooltip = (
+    tooltipHostProps: ITooltipHostProps,
+    defaultRender?: IRenderFunction<ITooltipHostProps>
+  ): JSX.Element => {
+    return <span className={tooltipHostProps.hostClassName}>{tooltipHostProps.children}</span>;
+  };
+
+  private _onColumnClick(column: IColumn, ev: React.MouseEvent<HTMLElement>): void {
+    const { onColumnClick } = this.props;
+    if (column.onColumnClick) {
+      column.onColumnClick(ev, column);
+    }
+    if (onColumnClick) {
+      onColumnClick(ev, column);
+    }
+  }
+
+  private _getColumnDragDropOptions(): IDragDropOptions {
+    const { columnIndex } = this.props;
+    const options = {
+      selectionIndex: columnIndex,
+      context: { data: columnIndex, index: columnIndex },
+      canDrag: () => this.props.isDraggable!,
+      canDrop: () => false,
+      onDragStart: this._onDragStart,
+      updateDropState: () => undefined,
+      onDrop: () => undefined,
+      onDragEnd: this._onDragEnd
+    };
+    return options;
+  }
+
+  private _onDragStart(item?: any, itemIndex?: number, selectedItems?: any[], event?: MouseEvent): void {
+    if (itemIndex && this.props.setDraggedItemIndex) {
+      this.props.setDraggedItemIndex(itemIndex);
+      this._root.current.classList.add(styles.borderWhileDragging);
+    }
+  }
+
+  private _onDragEnd(item?: any, event?: MouseEvent): void {
+    if (this.props.setDraggedItemIndex) {
+      this.props.setDraggedItemIndex(-1);
+      this._root.current.classList.remove(styles.borderWhileDragging);
+    }
+  }
+
+  private _onColumnContextMenu(column: IColumn, ev: React.MouseEvent<HTMLElement>): void {
+    const { onColumnContextMenu } = this.props;
+    if (column.onColumnContextMenu) {
+      column.onColumnContextMenu(column, ev);
```

Comment:
Just curious, why we need both column.onColumnContextMenu and props.onColumnContextMenu. We'll end up executing both if they are provided, not sure if that is intentional too.

Reply:
This is an existing function which was there in ColumnHeader, i just moved it as is.

Reply:
Ah I see... thanks!

---
In reply to: [196748992](https://github.com/OfficeDev/office-ui-fabric-react/pull/4857#discussion_r196748992) [](ancestors = 196748992)